tabpy-server/tabpy_server/app/util.py

```diff
@@ -35,3 +37,58 @@ def validate_cert(cert_file_path):
     if now > not_after:
         log_and_raise(https_error +
                       'The certificate provided expired on {}.'.format(not_after), RuntimeError)
+
+
+def parse_pwd_file(pwd_file_name):
+    '''
+    Parses passwords file and returns set of credentials.
+
+    Parameters
+    ----------
+    pwd_file_name : str
+        Passwords file name.
+
+    Returns
+    -------
+    succeeded : bool
+        True if specified file was parsed successfully.
+        False if there were ane issues with parsing specified file.
+
+    credentials : dict
+        Credentials from the file. Empty if succeeded is False.
```

Comment:
It looks like this may not be true if the first few entries are valid but the Nth entry is a dupe (for instance).  need to test this assumption

Reply:
Line 89 is `return False, {}` where `{}` is an empty credentials collection